components/raftstore/src/store/peer.rs

```diff
@@ -1310,9 +1310,61 @@ impl Peer {
             ctx.current_time.replace(monotonic_raw_now());
         }
 
-        // The leader can write to disk and replicate to the followers concurrently
-        // For more details, check raft thesis 10.2.1.
         if self.is_leader() {
+            if let Some(hs) = ready.hs() {
+                // Correctness depends on the fact that the leader lease must be suspected before
+                // other followers know the `PrepareMerge` log is committed, i.e. sends msg to others.
+                // Because other followers may complete the merge process, if so, the source region's
+                // leader may get a stale data.
+                //
+                // Check the committed entries.
+                // TODO: It can change to not rely on the `committed_entries` must have the latest committed entry
+                // and become O(1) by maintaining these not-committed admin requests that changes epoch.
+                if hs.get_commit() > self.get_store().committed_index() {
+                    assert_eq!(
```

Comment:
Can be changed to `debug_assert`.

Reply:
I think it's necessary to `assert` because it's the correctness dependency.